### PR TITLE
fix out of range call to block's receipts

### PR DESCRIPTION
**Pull request by winsvega**
configure blockchain tests mining
make all fields hex and prefixed with 0x

related test update:
https://github.com/ethereum/tests/pull/154

### CODE REVIEW

File: libethereum/Block.h [hunk 1]
```diff
@@ -191,16 +191,16 @@ class Block
 	h256Hash const& pendingHashes() const { return m_transactionSet; }
 
 	/// Get the transaction receipt for the transaction of the given index.
-	TransactionReceipt const& receipt(unsigned _i) const { return m_receipts[_i]; }
+	TransactionReceipt const& receipt(unsigned _i) const { return m_receipts.at(_i); }
```
Comment (gumb0) [inline libethereum/Block.h]: Is there a catch for `std::out_of_range` or at least `std::exception` everywhere where it's called?

Comment (winsvega) [inline libethereum/Block.h]: there is a catch somewhere - it will just print an error message to the error log.

Comment (chfast) [inline libethereum/Block.h]: In case of JSON RPC, shouldn't we response with a JSON error, not with command line log?

**Comment by winsvega** [comment]
that bug was discovered during test framework changes. so...
regarding to RPC we'll have to look at the behaviour when sending an actual RPC request.

File: libethashseal/Ethash.cpp [hunk 2]
```diff
@@ -163,7 +163,7 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 
 	// Avoid transactions that would take us beyond the block gas limit.
 	u256 startGasUsed = _env.gasUsed();
-	if (startGasUsed + (bigint)_t.gas() > _env.gasLimit())
+	if (startGasUsed + _t.gas() > _env.gasLimit())
```
Comment (chriseth) [inline libethashseal/Ethash.cpp]: We have to guard against overflow somehow.

**Pull request merged**